crates/libs/registry/src/key.rs

```diff
@@ -320,14 +327,63 @@ impl Key {
         }
     }
 
+    /// Sets the name and value in the registry key.
+    #[cfg(feature = "std")]
+    pub fn set_os_string<N: AsRef<str>, V: AsRef<std::ffi::OsStr>>(
+        &self,
+        name: N,
+        value: V,
+    ) -> Result<()> {
+        let value = encode_wide(value);
+
+        unsafe { self.set_value(name, REG_SZ, value.as_ptr() as _, value.len() * 2) }
+    }
+
+    /// Gets the value for the name in the registry key.
+    #[cfg(feature = "std")]
+    pub fn get_os_string<T: AsRef<str>>(&self, name: T) -> Result<std::ffi::OsString> {
```

Comment:
This would be a lot more efficient if `OsString` were an `HSTRING` but I'm guessing its too late for that now. 🥲

Reply:
Sadly, that ship set sail in 2015...

I do have a PR I need to get back to that will (hopefully) allow passing wide strings to filesystem APIs. Return types are stuck with `OsString` though.